### ISSUE TITLE
Warnings when training the modified model on the first round.

### ISSUE THREAD

**Issue by weir12** [nanoporetech/taiyaki#32]
Hi:
When I run the following command：
`train_mod_flipflop.py --device 0 --mod_factor 0.01 --outdir training taiyaki/models/mGru_cat_mod_flipflop.py modbase.hdf5`
The following warning appears
```
......* Warning: only 47 chunks passed tests after 108 attempts.
* Summary: pass:47 meandwell:17 maxdwell:44
..* Warning: only 56 chunks passed tests after 114 attempts.
* Summary: pass:56 meandwell:14 maxdwell:43 emptysequence:1
...........* Warning: only 59 chunks passed tests after 126 attempts.
* Summary: pass:59 maxdwell:53 meandwell:13 emptysequence:1
.....* Warning: only 51 chunks passed tests after 108 attempts.
* Summary: pass:51 maxdwell:42 meandwell:15
...* Warning: only 46 chunks passed tests after 116 attempts.
* Summary: pass:46 maxdwell:51 meandwell:19
..* Warning: only 43 chunks passed tests after 102 attempts.
* Summary: pass:43 maxdwell:48 meandwell:11
.......* Warning: only 46 chunks passed tests after 100 attempts.
* Summary: pass:46 maxdwell:46 meandwell:8
...* Warning: only 38 chunks passed tests after 102 attempts.
* Summary: pass:38 maxdwell:49 meandwell:15
.   998 0.090  138.71s (70.14 ksample/s 2.37 kbase/s) lr=4.00e-04 pass:3412193  maxdwell:1954485  meandwell:647545  emptysequence:25253  tooshort:58
......* Warning: only 46 chunks passed tests after 106 attempts.
* Summary: pass:46 maxdwell:47 meandwell:13
............* Warning: only 45 chunks passed tests after 102 attempts.
* Summary: pass:45 maxdwell:42 meandwell:15
................* Warning: only 51 chunks passed tests after 108 attempts.
* Summary: pass:51 maxdwell:46 meandwell:11
.............* Warning: only 52 chunks passed tests after 114 attempts.
* Summary: pass:52 maxdwell:46 meandwell:16
...   999 0.090  134.75s (73.59 ksample/s 2.48 kbase/s) lr=4.00e-04 pass:3415745  maxdwell:1956435  meandwell:648244  emptysequence:25283  tooshort:58
..................* Warning: only 47 chunks passed tests after 102 attempts.
* Summary: pass:47 meandwell:8 maxdwell:47
.* Warning: only 48 chunks passed tests after 102 attempts.
* Summary: pass:48 meandwell:9 maxdwell:45
..........* Warning: only 46 chunks passed tests after 104 attempts.
* Summary: pass:46 maxdwell:52 meandwell:6
....* Warning: only 48 chunks passed tests after 104 attempts.
* Summary: pass:48 maxdwell:48 meandwell:8
.* Warning: only 50 chunks passed tests after 104 attempts.
* Summary: pass:50 maxdwell:48 meandwell:6
..............* Warning: only 59 chunks passed tests after 122 attempts.
* Summary: pass:59 maxdwell:51 meandwell:12
.C  1000 0.090  137.18s (72.25 ksample/s 2.43 kbase/s) lr=4.00e-04 pass:3419207  maxdwell:1958427  meandwell:648857  emptysequence:25302  tooshort:58
```
How to understand the meaning of **chunk**？Does it mean something like ‘batch size’?
Thank you for your support.

**Comment by lpryszcz** [comment]
Are you training DNA or RNA models?

**Comment by aevansNP** [comment]
We use the word 'chunk' to mean a segment of data used for training.
The batch size is the number of chunks used in a single iteration of the training loop.
For example, if the batch size is 100, then in a single training step, the training script chooses 100 chunks.

Each chunk is chosen by selecting a random read from the training set and then choosing a random segment of length (say) 2000 current samples in that read. The training data also contains the sequence data that goes with the samples. That data (samples + bases that go with them) is a chunk.

Chunks must pass some tests (filters) before they are used in training. A chunk is rejected if the maximum dwell time (i.e. number of samples representing a single base) or mean dwell time (average samples per base) is outside a range determined by sampling done at the start of training.

These warning messages tell you that more chunks than expected are failing these filter tests. The reason could be that something has gone wrong in the biochemistry, or it could just be that the conditions in whatever experiment you are doing are more variable than in the tests we did to set up the code.

Training will still proceed with these warnings - it looks like about half the training data is failing the tests. You have the choice of carrying on as things are (excluding some of your training data, which will probably make training proceed more quickly because of lower data variability, but might mean that the more extreme data in your test set is basecalled less well), or relaxing the filter parameters to include the whole test set.

You might find the script misc/plot_mapped_signal_files.py helpful - you can use it to inspect the mappings in your training data file and see whether (for example) some of them look like they are particularly noisy or corrupted.

**Comment by weir12** [comment]
I'm trying train a RNA Modified model @lpryszcz

**Comment by tmassingham-ont** [comment]
The default parameters for training are set for DNA and are not appropriate for RNA.  https://github.com/nanoporetech/taiyaki/issues/23#issuecomment-498248344

Until we get the documentation updated, the following table should give guidance as to appropriate settings.

| Condition                | chunk\_len\_min | chunk\_len\_max | size | stride | winlen |
|--------------------------|-----------------|-----------------|------|--------|--------|
| DNA, high accuracy       |   2000          |   4000          | 256  | 2 or 3 | 19     |
| DNA, fast                |   2000          |   4000          | 96   | 4      | 19     |
| RNA, high accuracy       |   2000          |   4000          | 256  | 10     | 31     |
| RNA, fast                |   2000          |   4000          | 96   | 12     | 31     |

**Comment by weir12** [comment]
Thanks,I will follow your advice and re-train my model.

**Comment by callumparr** [comment]
relating to this. I am trying to train a model following the walkthrough but for RNA. 

In this case the first round builds upon a DNA modification model I presume and there is not intermediate modification model for RNA. How should one proceed? by using the Gmu model and then flagging the above parameters above for RNA?
